web3/middleware/pythonic.py

```diff
@@ -136,10 +136,10 @@ def to_hexbytes(num_bytes, val):
 
 
 LOG_ENTRY_FORMATTERS = {
-    'blockHash': apply_formatter_if(to_hexbytes(32), is_not_null),
-    'blockNumber': apply_formatter_if(to_integer_if_hex, is_not_null),
-    'transactionIndex': apply_formatter_if(to_integer_if_hex, is_not_null),
-    'transactionHash': apply_formatter_if(to_hexbytes(32), is_not_null),
+    'blockHash': apply_formatter_if(is_not_null, to_hexbytes(32)),
```

Comment:
@carver not necessarily right for this issue, but I'm thinking that we should `curry` the `apply_formatter_if` function and then define a bunch of these.

```python
apply_formatter_if_not_null = apply_formatter_if(is_not_null)
apply_formatter_if_integer = apply_formatter_if(is_integer)
```

Which cleans this up even further

```python
{
    'blockNumber': apply_formatter_if_not_null(to_integer_if_hex),
    ...
}
```

Reply:
Totally. Does the word `formatter` add much information here? It might be just as clear to use `apply_if_not_null`.